src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,165 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final Object[] EMPTY_DATAARRAY = {};
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] dataArray;
+    private int size;
+
+    public ArrayList() {
+        this.dataArray = new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        if (capacity == 0) {
+            this.dataArray = EMPTY_DATAARRAY;
+        } else if (capacity < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Capacity can't be negative");
+        } else {
+            this.dataArray = new Object[capacity];
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        growIfFull();
+        addOnNextFreeSell(value);
     }
```

Comment:
as Oksana mentioned, you are calling `growIfFull()` 2 times, here, and in the `addOnNextFreeCell()`, so we need to remove it from here, but then what we have left here, is just a call to `addOnNextFreeCell()`, which also not looking good, so we can just remove that private `addOnNextFreeCell()` method, and move its whole logic to the `add(T value)`, and call this public `add(T value)` method in other places instead of `addOnNextFreeCell()`, which would make total sense if you think about it, because we do not always need a common logic to be a private method, we can freely use public methods like in that example